bin/test-lxd-network-ovn

```diff
@@ -1400,6 +1402,90 @@ ovn_dhcp_reservation_tests() {
     lxc network delete lxdbr0
 }
 
+ovn_nested_vlan_tests() {
+    lxc network create lxdbr0 \
+        ipv4.address=10.10.10.1/24 ipv4.nat=true \
+        ipv4.dhcp.ranges=10.10.10.2-10.10.10.199 \
+        ipv4.ovn.ranges=10.10.10.200-10.10.10.254 \
+        ipv6.address=fd42:4242:4242:1010::1/64 ipv6.nat=true \
+        ipv6.ovn.ranges=fd42:4242:4242:1010::200-fd42:4242:4242:1010::254
+
+    # Create OVN network.
+    lxc network create ovn1 --type=ovn \
+        network=lxdbr0 \
+        ipv4.address=10.10.11.1/24 ipv4.nat=true \
+        ipv6.address=fd42:4242:4242:1011::1/64 ipv6.nat=true \
+        ipv6.dhcp.stateful=true
+
+    sleep 2
+    lxc network list | grep ovn1
+    lxc launch "${instanceImage}" u1 -s default -n ovn1 -c security.nesting=true
+
+    # Add nested NICs to u1.
+    lxc config device add u1 eth1 nic \
+        network=ovn1 \
+        nested=eth0 \
+        vlan=1 \
+        ipv4.address=10.10.11.2 \
+        ipv6.address=fd42:4242:4242:1011::2
+
+    eth1MAC="$(lxc config get u1 volatile.eth1.hwaddr)"
+
+    lxc config device add u1 eth2 nic \
+        network=ovn1 \
+        nested=eth0 \
+        vlan=2 \
+        ipv4.address=10.10.11.3 \
+        ipv6.address=fd42:4242:4242:1011::3
+
+    eth2MAC="$(lxc config get u1 volatile.eth2.hwaddr)"
+
+    lxc launch "${instanceImage}" u2 -s default -n ovn1
+
+    # Configure nested VLAN 1 inside u1 in a network namespace.
+    lxc exec u1 -- ip link add link eth0 name eth0.1 address "${eth1MAC}" type vlan id 1
+    lxc exec u1 -- ip netns add vlan_nesting1
+    lxc exec u1 -- ip link set eth0.1 netns vlan_nesting1
+    lxc exec u1 -- ip netns exec vlan_nesting1 ip link set lo up
+    lxc exec u1 -- ip netns exec vlan_nesting1 ip address add 10.10.11.2/24 dev eth0.1 nodad
```

Comment:
`nodad` doesn't seem to be ignored when adding IPv4.

Reply:
What do you mean?

Reply:
Sorry, I should have provided you the output:

```
# ip address add 10.10.11.2/24 dev eth0 nodad
Warning: nodad option can be set only for IPv6 addresses
```

Reply:
Ah it worked here, probably just a warning but ill remove for ipv4 for good measure. Just was easier to avoid forgetting to add it and getting flaky tests.

Reply:
Oh right, it does work but I stopped at the warning.